### ISSUE TITLE
Invalid Unused Field Warning

### ISSUE THREAD

**Issue by markvincentcaro** [golang/lint#433]
Sample Code:

**_dog.go_**

```
package main

type Dog struct {
    name string
}
```

**_main.go_**

```
package main

import (
    "fmt"
)

type Dalmatian Dog

func main() {
    var dalmatian Dalmatian
    dalmatian.name = "spot"
    fmt.Println(dalmatian)
}
```

**Expected Result:** Because Dog.name is used by Dalmatian, the field should not be flagged as unused.

**Actual Result:** Dog.name is incorrectly flagged as unused even though it is being used by Dalmatian.

**_Warning Text:_**
```
[go-lint] `name` is unused (structcheck)
```

**Environment:** Windows 10 64-bit

**IDE:** Visual Studio Code

**go-lint version:** Latest as of February 1, 2019.

**Comment by sedhossein** [comment]
same here

**Comment by ainar-g** [comment]
That's a warning from [`structccheck`](https://github.com/opennota/check/blob/master/cmd/structcheck/structcheck.go), not `golint`.  And please, when you use `golangci-lint` and alike, mention that in the issue text.

**Comment by markvincentcaro** [comment]
@ainar-g Ah, my mistake. Because the warning said `[go-lint]` I assumed that it had something to do with `golint`. I'll move this issue now. Thanks!
